File: airesources/Java/hlt/ThrustMove.java
package hlt;

public class ThrustMove extends Move {

    private final int angleDeg;
    private final int thrust;

    public ThrustMove(final Ship ship, final int angleDeg, final int thrust) {
        super(MoveType.Thrust, ship);
        this.thrust = thrust;
        this.angleDeg = angleDeg;
    }

    public int getAngle() {
        return angleDeg;
    }

    public int getThrust() {
        return thrust;
    }
}

File: airesources/CSharp/Halite2/hlt/ThrustMove.cs
namespace Halite2.hlt
{
    public class ThrustMove : Move
    {

        private int angleDeg;
        private int thrust;

        public ThrustMove(Ship ship, int angleDeg, int thrust)
            : base(MoveType.Thrust, ship)
        {
            this.thrust = thrust;
            this.angleDeg = angleDeg;
        }

        public int GetAngle()
        {
            return angleDeg;
        }

        public int GetThrust()
        {
            return thrust;
        }
    }
}
